### PR TITLE
fix (#1082): correct NatsTestClient stream publisher

### LINKED ISSUE #1082: Bug: TypeError: FakeProducer.publish() got an unexpected keyword argument 'timeout'
Hi!

**Describe the bug**

My test does not pass. I get the error: `TypeError: FakeProducer.publish() got an unexpected keyword argument 'timeout'`

Traceback:

```text

    @override
    async def publish(  # type: ignore[override]
        self,
        message: SendableMessage = "",
        reply_to: str = "",
        correlation_id: Optional[str] = None,
        headers: Optional[Dict[str, str]] = None,
        **producer_kwargs: Any,
    ) -> Optional[DecodedMessage]:
        assert self._producer, NOT_CONNECTED_YET  # nosec B101
        assert self.subject, "You have to specify outgoing subject"  # nosec B101
    
        extra: AnyDict = {
            "reply_to": reply_to or self.reply_to,
        }
        if self.stream is not None:
            extra.update(
                {
                    "stream": self.stream.name,
                    "timeout": self.timeout,
                }
            )
    
>       return await self._producer.publish(
            message=message,
            subject=self.subject,
            headers=headers or self.headers,
            correlation_id=correlation_id,
            **extra,
            **producer_kwargs,
        )
E       TypeError: FakeProducer.publish() got an unexpected keyword argument 'timeout'

```

**How to reproduce**

```python
import uuid
from datetime import datetime
from typing import Literal

from faststream.nats import NatsBroker
from pydantic import BaseModel, Field

class NatsPublisher:

    def __init__(self, settings: NATSSettings, nats_broker: NatsBroker | None = None) -> None:
        """Initialize."""
        self._nats_broker = nats_broker or NatsBroker(settings.url)

        self._publisher_gen_started = self._nats_broker.publisher(
            "generation.started", stream=settings.stream
        )
        self._publisher_gen_ended = self._nats_broker.publisher(
            "generation.ended", stream=settings.stream
        )

    async def publish_event_generation_started(
        self, payload: GenerationStartedEventModel
    ) -> None:
        """Send generation started event."""
        await self._publisher_gen_started.publish(BaseMessage(content=payload.model_dump()))

    async def publish_event_generation_ended(
        self, payload: GenerationEndedEventModel
    ) -> None:
        """Send generation ended event."""
        await self._publisher_gen_ended.publish(BaseMessage(content=payload.model_dump()))

```

Test:
```python
import pytest
from faststream.nats import NatsBroker, TestNatsBroker

@pytest.mark.asyncio()
async def test_given_nats_publisher_when_publishing_gen_start_event_then_it_works():
    # Given
    nats_settings = NATSSettings()
    broker = NatsBroker()

    @broker.subscriber("generation.started")
    def some_subscriber(msg: BaseMessage) -> None:
        _ = msg

    some_message = "some_message"

    async with TestNatsBroker(broker) as br:
        nats_publisher = NatsPublisher(nats_settings, br)

        # When
        await nats_publisher.publish_event_generation_started(some_message)  

        # Then
        assert some_subscriber.mock
        some_subscriber.mock.assert_called_once_with(some_message)
```

**Environment**
Running FastStream 0.3.8 with CPython 3.11.5 on Darwin

If the code above is not good enough to find the bug, I can write a better minimal reproducible example.

Thanks!

### PR DESCRIPTION
**Pull request by Lancetnik**
# Description

Please include a summary of the change and specify which issue is being addressed. Additionally, provide relevant motivation and context.

Fixes #1082

## Type of change

Please delete options that are not relevant.

- [ ] Documentation (typos, code examples, or any documentation updates)
- [x] Bug fix (a non-breaking change that resolves an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would disrupt existing functionality)
- [ ] This change requires a documentation update

## Checklist

- [x] My code adheres to the style guidelines of this project (`scripts/lint.sh` shows no errors)
- [x] I have conducted a self-review of my own code
- [ ] I have made the necessary changes to the documentation
- [x] My changes do not generate any new warnings
- [x] I have added tests to validate the effectiveness of my fix or the functionality of my new feature
- [x] Both new and existing unit tests pass successfully on my local environment by running `scripts/test-cov.sh`
- [x] I have ensured that static analysis tests are passing by running `scripts/static-anaylysis.sh`
- [ ] I have included code examples to illustrate the modifications

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -47,10 +47,10 @@ classifiers = [
 dynamic = ["version"]
 
 dependencies = [
-    "anyio>=3.7.1,<4; python_version < '3.11'",
-    "anyio>=3.7.1,<5; python_version >= '3.11'",
+    "anyio>=3.7.1,<5",
```
Comment (davorrunje) [inline pyproject.toml:50]: this will not work for Python bellow 3.11

**Review by davorrunje — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (Lancetnik) [reply pyproject.toml:50]: Why not? I tested locally and it works fine

Comment (Lancetnik) [reply pyproject.toml:50]: Btw, we was working with 4+ in the past, but with ExceptionGroup it brokes. For now your _compat works fine and the problem solved

Comment (davorrunje) [reply pyproject.toml:50]: I tested locally with anyio version 5.x and Python 3.10 and it did not work

Comment (Lancetnik) [reply pyproject.toml:50]: But it is limitation for 3.X and 4.X only

Comment (Lancetnik) [reply pyproject.toml:50]: What problem did you have?

Comment (davorrunje) [reply pyproject.toml:50]: from https://anyio.readthedocs.io/en/stable/versionhistory.html

4.0.0

BACKWARDS INCOMPATIBLE Replaced AnyIO’s own ExceptionGroup class with the PEP 654 BaseExceptionGroup and ExceptionGroup

BACKWARDS INCOMPATIBLE Changes to cancellation semantics:

- Any exceptions raising out of a task groups are now nested inside an ExceptionGroup (or BaseExceptionGroup if one or more BaseException were included)

- Fixed task group not raising a cancellation exception on asyncio at exit if no child tasks were spawned and an outer cancellation scope had been cancelled before

- Ensured that exiting a TaskGroup always hits a yield point, regardless of whether there are running child tasks to be waited on

- On asyncio, cancel scopes will defer cancelling tasks that are scheduled to resume with a finished future

- **On asyncio and Python 3.9/3.10, cancel scopes now only suppress cancellation exceptions if the cancel message matches the scope**

- Task groups on all backends now raise a single cancellation exception when an outer cancel scope is cancelled, and no exceptions other than cancellation exceptions are raised in the group

Comment (Lancetnik) [reply pyproject.toml:50]: It doesn't look like a change that could break our code.
Let me check again

Comment (davorrunje) [reply pyproject.toml:50]: I had problems with it, but please check. Maybe I was doing something wrong.

Comment (Lancetnik) [reply pyproject.toml:50]: @davorrunje as you can see, specific test is passing too

Comment (Lancetnik) [reply pyproject.toml:50]: This [condition](https://github.com/airtai/faststream/blob/main/faststream/_compat.py#L198) provides us with a py3.11< compatibility
All other stuff worked correctly already, so there is no any problem to use anyio4.x with any python version

**Review by davorrunje — APPROVED** [review]

**Pull request merged**